src/main/java/net/sf/jabref/logic/citationstyle/CitationStyleGenerator.java

```diff
@@ -42,21 +42,17 @@ public static String generateCitation(BibEntry entry, String style) {
      */
     public static String generateCitation(BibEntry entry, String style, CitationStyleOutputFormat outputFormat) {
         try {
-            String parsedEntry = new UnicodeToLatexFormatter().format(entry.toString());
-            InputStream stream = new ByteArrayInputStream(parsedEntry.getBytes(StandardCharsets.UTF_8));
+            BibTeXEntry bibTeXEntry = new BibTeXEntry(new Key(entry.getType()), new Key(entry.getCiteKey()));
+            for (Map.Entry<String, String> field : entry.getFieldMap().entrySet()) {
+                String value = UNICODE_TO_LATEX_FORMATTER.format(field.getValue());
+                bibTeXEntry.addField(new Key(field.getKey()), new DigitStringValue(value));
+            }
 
-            BibTeXDatabase db = new BibTeXConverter().loadDatabase(stream);
-            BibTeXItemDataProvider provider = new BibTeXItemDataProvider();
-            provider.addDatabase(db);
-
-            CSL citeproc = new CSL(provider, style);
-            citeproc.setOutputFormat(outputFormat.format);
-            provider.registerCitationItems(citeproc);
-
-            Bibliography bibliography = citeproc.makeBibliography();
+            CSLItemData cslItemData = new BibTeXConverter().toItemData(bibTeXEntry);
```

Comment:
Is it really required to instantiate BibTeXConverter each time? - The `UNICODE_TO_LATEX_FORMATTER` is also instantiated only once.